src/groupeddataframe/splitapplycombine.jl

```diff
@@ -451,6 +486,15 @@ wrap(x::AbstractMatrix) =
     NamedTuple{Tuple(gennames(size(x, 2)))}(Tuple(view(x, :, i) for i in 1:size(x, 2)))
 wrap(x::Any) = (x1=x,)
 
+wrap_table(x::Any) =
```

Comment:
@nalimilan - in the end we yet need a different approach to `wrap_table`.

Reply:
I would leave it for later (when we add pseudo-broadcasting in `combine`) as then we will have to refactor it yet again.

The current logic for "row" handling is the following (I have not written it, but I want to minimize changes in this PR):
1. In `_combine` we call `idx, outcols, _ = _combine_with_first(wrap(firstres), fun, gd, incols)` where `wrap` should accept anything (as we do not know yet what we will get - tables or rows)
2. then if we go for row-version `_combine_rows_with_first!` we use `fill_row!` which checks if we do not mix scalars and vectors

so we are safe here.

Reply:
OK. But since you move the check from `_combine_table_with_first!` to `wrap_table`, it would make the code more consistent to also move the check from `_combine_rows_with_first!` to `wrap_rows`.

Reply:
OK. I introduced `wrap_row` then. It is (mildly) tricky, so a careful look is welcome. This change indeed might improve performance a bit (we remove two checks in `fill_row!` which I think were not always optimized-out)